src/lib/expansion/expansion-panel-header.ts

```diff
@@ -101,6 +102,17 @@ export class MatExpansionPanelHeader implements OnDestroy {
     .subscribe(() => this._changeDetectorRef.markForCheck());
 
     _focusMonitor.monitor(_element.nativeElement, renderer, false);
+
+    // When a panel is hosted in an accordion, subscribe to the expand/collapse subject
+    if (this.panel.accordion) {
```

Comment:
By placing the logic for setting panel expansion state in the `MatExpansionPanelHeader` it ties the logic specifically to the Material implementation rather than the CDK base.

I would suggestion adding this logic to the `AccordionItem` class.

There you can set the expanded state directly.

Reply:
I need access to the `disabled` state of the hosting panel to prevent disabled panels from expanded/collapsed, which is why I used `_toggle()`. Is there another way to get access to this property without too much refactoring?

Reply:
Think I may have solved it with `mixinDisabled` - let me know if that is not a desired solution though. Cheers.

Reply:
We actually just removed the mixinDisabled from AccordionItem as we are moving it to CDK.  You can see the move in #7530 

I supposed you could bake the `expandAll`/`collapseAll` into the `AccordionItem`, and then in `MatExpansionPanel`, extend the setter of `expanded` to respect `disabled`.  This would allow the `expandAll`/`collapseAll` functionality to live in the CDK, but work as expected in `MatAccordion`.  Admittedly I haven't thought this idea entirely through and would need to consider the ramifications a bit more before saying its definitely the best option or the option to go with.

Reply:
Thanks, @josephperrott - I will wait for 7530 to be merged and then take it from there.